sse/src/main/scala/akka/stream/alpakka/sse/scaladsl/EventSource.scala

```diff
@@ -94,11 +94,9 @@ object EventSource {
         }
         send(request).flatMap(Unmarshal(_).to[EventSource]).fallbackTo(noEvents)
       }
-      def recover(eventSource: EventSource) = eventSource.recoverWithRetries(1, { case _ => Source.empty })
-      def delimit(eventSource: EventSource) = eventSource.concat(singleDelimiter)
-      Flow[Option[String]]
-        .mapAsync(1)(getEventSource)
-        .flatMapConcat((recover _).andThen(delimit))
+      def recoverAndDelimit(eventSource: EventSource) =
+        eventSource.recoverWithRetries(0, { case _ => Source.empty }).concat(singleDelimiter)
+      Flow[Option[String]].mapAsync(1)(getEventSource).flatMapConcat(recoverAndDelimit)
```

Comment:
If you use `recoverWithRetries(0, {... whatever ...})` that will always fail the stream and never retry, because of the retry `0`. Try this one out for example:

```
Source.failed(new RuntimeException("bork")).recoverWithRetries(0, {
    case _: Throwable =>
      println("recovering")
      Source.empty
  }).runForeach(println)
    .onComplete(t => println("done: " + t))
```

You should see it always fails the stream with the `RuntimeException`, so that is the same as just not using `recoverWithRetries` at all, if you want it to recover with the given source the value of `attempts` must be `1` (or higher, that doesn't matter as `Source.empty` will never fail)